### ISSUE TITLE
Keep loosing edited files and updates created in play.nativescript.org after closing browser.

### ISSUE THREAD

**Issue by mrwrighty** [NativeScript/playground-feedback#40]
I keep loosing changes and edits made on play.nativescript.org.

I did a lot of work today, obviously saving the changes and testing via QR code on Nativescript Preview.

I then closed down my browser to reboot after some OSX updates. Once logged back in I restarted my browser went to play.nativescript.org and all my changes have gone. Even the NPM plugins for Google Maps has disappeared. Also folder with new files I created are also missing.

This isn't the first time I have noticed this. I moved between a MacBook Pro and a Windows Laptop and notices that files that I was editing on the MAC were not visible on the Windows laptop.

I'm very annoyed as I have lost a lot of work. I did download a copy to test under SideKick which is complete and up to date, but that version is not showing on the internet projects with the same up to date content. Whats gone wrong or going wrong.

**Comment by vhristov5555** [comment]
First sorry for the inconvenience caused.
Could you please explain what do you mean by saving:
- cmd(ctrl) + s or Preview button
- Click the Save button

cmd(ctrl) + s or Preview button is doing a preview. The changed files in you project are transferred to the preview app.

The saved button is actually making a save. We are going to generate a new link, which you could open on every browser and platform and all changes are going to be saved. Of course you will need to scan the QR code again when you open the saved link. 
![Screen Shot 2018-05-31 at 11.09.09.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59e45dd4eec5e00011963e48/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77266457f26b7191a379ee0c2b58415f47be5d4a1f06a3a3dc78e2a436628f555ee9d53e5a77c56844faec861fbf09e920dd1b83e4b89850d2aa6c7bb3d4d216744b470a83ed5c3caa984eca5ff14ee54e5cac143565f43b.png)
Also the saved session will be shown in projects tab(you need to be logged in to see it). Where you can see all of your project's versions. You can delete or open a previous version.
![Screen Shot 2018-05-31 at 11.09.58.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59e45dd4eec5e00011963e48/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77266457f26b7191a379ee0c2b58415a46be5d4a1f06a3a3dc78e2a436628f555ee9d53e5a77c56844faec861fbf09e920dd1b83e4b89850d2aa6c7bb3d4d216744b470a83ed5c32aa984eca5ff14ee54951a01a346bf43b.png)

I hope this information is useful to you. Also I hope I have answered your question.

Please don't hasted to write to us if you have further problems.

Best Regards,
Vasil Hristov

**Comment by mrwrighty** [comment]
Hi Thanks for your response. I am clicking the save menu option.

I have attached a video showing the problem. You will see me add an image to the project. Save it and the image appears on the device via preview. If I then close my browser session and re-open it. you will see that although I saved the project, the image is missing from the code when I re-open the project. I am wary of closing the browser when working in play.nativescript.org for fear of loosing work.
[Nativescript2.mp4.zip](https://github.com/NativeScript/playground-feedback/files/2057469/Nativescript2.mp4.zip)

**Comment by vhristov5555** [comment]
Hi,
Thanks for the feedback. I tried to reproduce the issue multiple times but i was not able to do it.
Could I ask you several questions:
- What browsers and versions this issue is reproducible in?
- Could please send us the saved session from the video or some session in which you are able to reproduce the issue?
- Could you please verify that the issue is always reproducible on your side or it happens from time to time ?
- What I could see in the video is that the project which you are using is created 8 minutes ago. Does that mean that you have stop the video after the save and continue the recording after that or the saved procedure has failed?
![Screen Shot 2018-05-31 at 15.25.03.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59e45dd4eec5e00011963e48/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77266457f26b7191a379ea0c2954415f4dbe5d4a1f06a3a3dc78e2a436628f555ee9d53e5a77c56844faec861fbf09e920dd1b83e4b89850d2aa6c7bb3d4d216744b470a83ed5c3caa984eca5ff14ce14555a7123167f43b.png)
- Could please check if any error is logged in the developer console?
- Could you please verify if the version in the generated URL has increased when your save is failing? For example, if this is the saved session https://play.nativescript.org/?template=play-ng&id=DLO1Iq&v=2 and you make a change and save the version should increase to 3 in the URL.

**Comment by mrwrighty** [comment]
Hi 

Browser is Safari 11.1 on OSX 10.13.4

Save is not incrementing the version number, the test project I used still has only a single version. As you saw in the video I saved it but not version number was generated. Equally no share URL option appeared. This is also evident in the video after the save, no share URL option appears.

The Video was not stopped or edited. What you see is me starting with the test project modifying and saving. Closing the browser tab and opening the project in a new tab to show the missing edits. For the record I have lost my work again this morning after I copied the text content from downloaded copy and pasted into the play.nativescript.org version.

**Comment by vhristov5555** [comment]
Okay sounds like some kind of caching issue. 
Could you please try if this behaviour is reproducible in new private window?

**Comment by mrwrighty** [comment]
No same problem. I have reset all cached data and Chrome was the first time I had used this browser to access to projects. A New Private windows in Safari resulted in same problem. Test project is now on V11 but project manager is still only showing V1

**Comment by vhristov5555** [comment]
I have tried to reproduce the issue multiple times with different configurations, but I was not able to do it. What I could say is that for some reason you are not getting the incremented versions. Sounds like this is some kind of caching on your side. 
I could recommend to try Google Chrome private browser. Also to try on different machine. And different networks.
I could recommend to try Nativescript Sidekick(https://www.nativescript.org/nativescript-sidekick).
Also NativeScript command-line interface(https://docs.nativescript.org/angular/tutorial/ng-chapter-1).
Both tools are recommend for develop Nativescript apps. 
Sorry for the inconvenience cause.

**Comment by mrwrighty** [comment]
Hi.

Sorry for the late reply but had a busy weekend. I'm still having the issue on my MAC but I have other devices which I can also test on. I have just tested the problem on Windows 10 Edge which has never previously been used for access to play.nativescript.org. I'm using a new VM.

The problem persists across Windows 10 as well, so cannot be a caching issue. My test app has incremented to V12, but the project manager is still only showing the original V1 file. The is no option to expand the project (Clicking arrow) to display the project history.

I have a number of projects and have noticed even in these the numbering is not sequential, missing some versions, so can only assume the issue has been happening for a while. For example I have one project on V2033, but the history shows v2025, v2018, v2006, v1985, v1945 etc in that order so as you can see large gaps in the recorded history.

I am using Sidekick for the final coding and compilation of apps, but very much like play.nativescript.org for the quick drag and drop coding and testing. As I'm new the NS it helps me learn and understand the controls and their definitions and options.

**Comment by vhristov5555** [comment]
Hi,

Could you please try this behaviour on the same machine that the problem occurs but with different profile. 
You need to log out and log in with different profile. 
May be we are facing some kind of limitation with so many saved session.

**Comment by marklanhamhc** [comment]
Ok, I've been testing this for the last hour and it seems the problem is two-fold. The project saves ok and gives me a URL (with version 160) however when I click on Projects its only showing up to version 153.

The other problem is that I can only create up to 6 SASS files in a folder at any time, any more than this, once I save and reopen the project it only shows the first 6.

**Comment by mrwrighty** [comment]
Hi, glad in some respects that its not just me. The only way round the problem that I could see was to note the count in the url after saving a project then manually change the URL when loading a project in the future to the last known version so it loads in the workspace. Its quite a serious bug as far as I'm concerned because until I realised what was happening I thought I had lost a ton of work and re did it. It was only when I went through my browser history did I understand how it was supposed to work but didn't in reality. Playing around with the V number in the URL allowed me to retrieve the lost work.

**Comment by vhristov5555** [comment]
Also if it possible could please send us the saved sessions. In such way we could more easily the caused of the issues you are facing.

Thanks in advance.

**Comment by mrwrighty** [comment]
For my current project I've moved away from play.nativescript.org to Nativescript SideKick and Brackets editor. I very much like the drag and drop design function of Playground as a starting point but got frustrated with its limitations for testing in the Playground App. Hopefully @marklanhamhc can send you some saved sessions.

**Comment by marklanhamhc** [comment]
Yes, I'm also working with another developer on the same project and he also found the same problem with Playground. For the main build we're using Sublime with Android and iOS emulators. However the Playground has been great for stakeholders to see the app on their phones, has worked really well up until now.

**Comment by vhristov5555** [comment]
We want to find the issue and to fix it, but this kind of issue are hard to find. I have created a session with 11 sass files and 580 saved sessions https://play.nativescript.org/?template=play-ng&id=MtaA5W&v=580
![image](https://user-images.githubusercontent.com/7846198/43588719-c6d199c8-9675-11e8-9e0c-dbcbdbf63053.png)
![image](https://user-images.githubusercontent.com/7846198/43588732-cd4d661a-9675-11e8-80b8-416603ad8f26.png)
All sessions are shown also the 11 sass files are visible in test folder.
![image](https://user-images.githubusercontent.com/7846198/43588783-e908d196-9675-11e8-8ae6-55194d11c671.png)
Could you please send us a saved session in which the problem is visible so we could get more information about it ?

Thanks in advance.

**Comment by marklanhamhc** [comment]
As this is a private project, whats the best way to share this with you? I don’t want to publish the link online?

**Comment by vhristov5555** [comment]
Could please give us the Email you are using to login in the account ?

**Comment by mrwrighty** [comment]
It is a project list level the problem occurs. Saving any modifications is fine and the URL V number increments as expected, but if you close the browser window down and then go back to the project and open it, this is when the problem occurs. The Project list is not showing the latest version to open which is why it looks like changes have gone missing.

Once the project is opened you then have to go to the URL, change the V number to the latest number and you then get the latest changes shown in your files.

**Comment by vhristov5555** [comment]
I have tried it in private browser:
![image](https://user-images.githubusercontent.com/7846198/43590336-6da45d6e-9679-11e8-824e-5449fc456109.png)
also in Safari:
![image](https://user-images.githubusercontent.com/7846198/43590346-741a4d48-9679-11e8-8ad7-12f310ceb76c.png)
So may be some specific files or number of files are causing the issue. Or it may be combination of projects that are used.

**Comment by marklanhamhc** [comment]
Thanks Vasil, I have emailed you directly.

**Comment by vhristov5555** [comment]
We have deploy a fix for this problem.

**Comment by immarina** [comment]
I don't know if you would consider this as related, but i was working on a project and saving it. I got the URL to be able to come back to it and it works perfectly, however in the middle of working on it I created an account and the project wont save into my "Projects" tab. If I go ahead and create a NEW project and save it, it does appear on my projects tab. It would seem the other project is getting owned by some unregistered user, and when I logged in/registered after having saved it it wont get saved into my own account.
